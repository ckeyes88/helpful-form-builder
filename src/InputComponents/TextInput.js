import React from 'react';
import { FormGroup, ControlLabel, FormControl, HelpBlock } from 'react-bootstrap';

export default class TextInput extends React.Component {

  render() {
    let validationState = this.props.getValidationState ? this.props.getValidationState(this.props.value) : {};
    return (
      <FormGroup
        controlId={this.props.label}
        validationState={validationState.state} >
        <ControlLabel >{this.props.label}</ControlLabel>
        <FormControl
          type="text"
          value={this.props.value}
          onChange={this.props.onChange}/>
        <FormControl.Feedback />
        <HelpBlock>{validationState.message}</HelpBlock>
      </FormGroup>
    );
  }
}

TextInput.propTypes = {
  value: React.PropTypes.oneOfType([React.PropTypes.string, React.PropTypes.number, React.PropTypes.bool]).isRequired,
  label: React.PropTypes.string.isRequired,
  getValidationState: React.PropTypes.func,
  isDisabled: React.PropTypes.func,
  onChange: React.PropTypes.func.isRequired
}
