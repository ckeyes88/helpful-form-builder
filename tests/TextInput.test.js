import React from 'react';
import TextInput from '../src/InputComponents/TextInput';
import renderer from 'react-test-renderer';
jest.mock('react-dom');

test("TextInput renders correctly", () => {
    const component = renderer.create(
        <TextInput 
            id={'text-field'}
            value={"Chip"}
            label={"First Name"}
            onChange={(ev) => {console.log(ev)}}/>
    );

    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
    console.log(tree);
    tree = component.toJSON();
    expect(tree).toMatchSnapshot();
})