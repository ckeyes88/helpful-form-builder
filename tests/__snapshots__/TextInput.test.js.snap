exports[`test TextInput renders correctly 1`] = `
<div
  className="form-group has-feedback">
  <label
    className="control-label"
    htmlFor="text-field">
    First Name
  </label>
  <input
    className="form-control"
    id="text-field"
    onChange={[Function]}
    type="text"
    value="Chip" />
  <span
    className="help-block" />
</div>
`;

exports[`test TextInput renders correctly 2`] = `
<div
  className="form-group has-feedback">
  <label
    className="control-label"
    htmlFor="text-field">
    First Name
  </label>
  <input
    className="form-control"
    id="text-field"
    onChange={[Function]}
    type="text"
    value="Chip" />
  <span
    className="help-block" />
</div>
`;
